diff --git a/utils/plots.py b/utils/plots.py
index 1bbb9c0..9222cbb 100644
--- a/utils/plots.py
+++ b/utils/plots.py
@@ -80,7 +80,7 @@ class Annotator:
             self.im = im
         self.lw = line_width or max(round(sum(im.shape) / 2 * 0.003), 2)  # line width
 
-    def box_label(self, box, label='', color=(128, 128, 128), txt_color=(255, 255, 255)):
+    def box_label(self, box, label='', color=(128, 128, 128), txt_color=(255, 255, 255), fps=''):
         # Add one xyxy box to image with label
         if self.pil or not is_ascii(label):
             self.draw.rectangle(box, width=self.lw, outline=color)  # box
@@ -110,6 +110,13 @@ class Annotator:
                             txt_color,
                             thickness=tf,
                             lineType=cv2.LINE_AA)
+                cv2.putText(self.im,
+                            fps, (550, 20),
+                            5,
+                            self.lw / 3,
+                            txt_color,
+                            thickness=1,
+                            lineType=cv2.LINE_8)
 
     def rectangle(self, xy, fill=None, outline=None, width=1):
         # Add rectangle to image (PIL-only)
